### PR TITLE
Optionally load the config via forRoot

### LINKED ISSUE #28: How to use app_id returned from API
Is it possible to provide the {app_id} for intercom at a later stage? I have to pull that using an API call so its not available at boot time.

For that reason i dont have app_id to put here:
```
IntercomModule.forRoot({
      appId: <your_app_id>, // from your Intercom config
})
```

### PR DESCRIPTION
**Pull request by sjroesink**
Currently `IntercomModule.forRoot` forces you to provide the config when loading the module. This prevents config from being provided asynchronous.

This PR makes it optional to provide the config initially via `IntercomModule.forRoot`. The function `Intercom.loadIntercom` can then be called at a later point.

This fixes #28

### CODE REVIEW

**Comment by scott-wyatt** [comment]
@wbhob thoughts on this?

File: src/app/ng-intercom/intercom/intercom.ts [hunk 1]
```diff
@@ -248,6 +237,8 @@ export class Intercom {
     //   return
     // }
 
+    this.config = config
```
Comment (wbhob) [inline src/app/ng-intercom/intercom/intercom.ts]: Why is this necessary?

Comment (sjroesink) [reply src/app/ng-intercom/intercom/intercom.ts]: Because the config (app_id) is later used in the `boot` function.

**Comment by scott-wyatt** [comment]
I'm closing this for now, I believe that most of this has been implemented in one form or another and the angular 6+ version is now available in master.

**Comment by nitinkg7** [comment]
Which version to use for making config optional?

**Comment by wbhob** [comment]
with forRoot, config is required because it registers your appId. Without the app ID, Intercom will not work.
